### DISCUSSION TITLE
Not getting orders

### DISCUSSION THREAD

**Discussion by mustajababbas** [AshleyRedman/WPClothes2Order#114] [Q&A]
Hey there,
why I'm not getting any orders in my clothe2order nor any order note in woocommerce order dashboard kindly have a look at SS attachment, I've checked the API by custom PHP code and it worked.
I don't have any idea what I'm getting wrong here

![Screenshot 2023-01-20 130509](https://user-images.githubusercontent.com/85496904/213647720-78432417-3615-4891-bf98-9577388f186e.png)

**Comment by AshleyRedman** [comment]
Hi @wawawasi,

There isn't a whole lot of information to go off from what you have said. Our plugin does allow you to view the raw response from C2O, by going to:

WPC2O plugin settings -> Order history -> find your order -> view order payload. 

You will see the data sent to C2O as well as their response. Feel free to also take a screenshot/copy that info and provide it here.

![Screenshot 2023-01-20 at 09 21 33](https://user-images.githubusercontent.com/7602057/213660746-f7dd9219-3c37-4422-8085-7724c92bd22d.png)

**Reply by mustajababbas** [reply]
I'm not getting anything here either 

![image](https://user-images.githubusercontent.com/85496904/213667905-b17eb336-4244-4398-b37c-d550653c9947.png)

**Reply by AshleyRedman** [reply]
Ok, so looks like no orders have actually been sent, so C2O has not rejected them, its that the order hasnt gone through. 

What does this screen look like for you?

Woocommerce -> orders 

Example:
![Screenshot 2023-01-20 at 11 00 12](https://user-images.githubusercontent.com/7602057/213680053-ab9a8782-37f2-4b2e-ac7c-b5064dc91dad.png)

**Reply by mustajababbas** [reply]
Have a look

![Screenshot 2023-01-20 161809](https://user-images.githubusercontent.com/85496904/213683270-f7dbcee4-c589-4b5e-be4d-c70f155df987.png)

**Reply by AshleyRedman** [reply]
Ok, so i think the order sending to C2O isn't being triggered.
Our plugin starts the sending order process when the customer lands on the thank you page, i.e the WordPress/WooCommerce thank you hook triggers. See https://github.com/AshleyRedman/WPClothes2Order/blob/493b4561a55b04171c3df0fb225e3701c65c2a92/wpclothes2order.php#L83

So, you may have a conflicting plugin(s), or have custom code which is preventing that hook from firing. 

The best way to debug this is to disable potentially conflicting plugins and test an order in test mode.

**Reply by mustajababbas** [reply]
Man I've disabled all plugins except WPClothes2Order and woocommerce, switched my theme to default theme and still no solution

**Reply by AshleyRedman** [reply]
Ok thanks for letting me know. 
It may be useful to add the two following plugins and retesting as it may show what the issue might be:

https://en-gb.wordpress.org/plugins/wp-debugging/
+ 
https://en-gb.wordpress.org/plugins/query-monitor/

Also, can you tell me what version of PHP you are running?
You can find that out by going to WordPress Admin -> Tools -> Site Health -> Info Tab -> Server.
Example would look like:

![Screenshot 2023-01-20 at 12 06 59](https://user-images.githubusercontent.com/7602057/213691483-e6803e45-b95c-4e4f-bcd2-9e60f052234c.png)

**Reply by mustajababbas** [reply]
This is my version i think its alright according to plugin requirements

![image](https://user-images.githubusercontent.com/85496904/213696237-8abc4baa-8fa1-40fe-a8e7-3743a05efb63.png)

**Reply by AshleyRedman** [reply]
Yep that all looks good. Can you confirm you're using the latest version of the plugin downloaded from https://wordpress.org/plugins/wpclothes2order/ ?

I would try activating the two plugins i mentioned here & see if they give any indication of the issue https://github.com/AshleyRedman/WPClothes2Order/discussions/114#discussioncomment-4736735

**Reply by mustajababbas** [reply]
Only getting this issue

![image](https://user-images.githubusercontent.com/85496904/213699416-13162eca-9a8e-42d2-9928-7f23977c6536.png)

**Reply by AshleyRedman** [reply]
Ok thank you. That looks good, the only message you have there is unrelated. 

Can you confirm that you have supplied valid entries on this screen?

![Screenshot 2023-01-20 at 13 01 38](https://user-images.githubusercontent.com/7602057/213701143-e8dd76da-8ecf-4fcd-8a57-58a3c8456371.png)

**Reply by mustajababbas** [reply]
I guess its okay

![image](https://user-images.githubusercontent.com/85496904/213702201-32d098e0-17f4-4a40-b939-7eeab632ec70.png)

**Reply by AshleyRedman** [reply]
Ok thank you, that looks good to me. 
I think the overall problem is that the thankyou hook is not being called. 

Can you try & enable 'Cash on delivery' payment option & test purchasing a product?
It may be that the thankyou hook is not being called on a particular payment method, so testing with cash-on-delivery might narrow that possibility down. 

![Screenshot 2023-01-20 at 13 15 57](https://user-images.githubusercontent.com/7602057/213703583-219de279-e5dc-45a7-95a3-f685ed1e79bb.png)

**Reply by mustajababbas** [reply]
I'm currently testing it on COD I might have to check on another payment method?

**Reply by AshleyRedman** [reply]
I am just about to release a patch for the plugin that updates the hook used when orders are placed:

https://github.com/AshleyRedman/WPClothes2Order/pull/115

Once version 1.0.9 has been released to the WordPress org, I will comment here. That released, can you update your plugin to the latest & see if that resolves the issue.

**Reply by AshleyRedman** [reply]
@wawawasi Version 1.0.9 has now been released: https://wordpress.org/plugins/wpclothes2order/
Please update your plugin to the latest version to see if this resolves the issue.

**Reply by mustajababbas** [reply]
I just updated it to 1.0.9 now it shows Internal Server Error on the place order button every time.
Working fine without WPClothes2Order thought ☹

**Reply by mustajababbas** [reply]
So Place order give me an internal server error but somehow order has been generated in my orders tabs 

![image](https://user-images.githubusercontent.com/85496904/213722998-0f73dc87-0667-4e6b-a16c-deef73500245.png)

**Reply by AshleyRedman** [reply]
Hmm, i am having problems reproducing the issue. I have a test site (https://wpclothes2order.ajrsoftware.com/) running PHP 7.4, using WPClothes2Order `1.0.9` and it is working ok. Not sure what else I can do from here. If you have WP_DEBUG set to `true` and you are getting any errors/warnings, then feel free to post here, but right now I cannot reproduce

**Reply by mustajababbas** [reply]
Alright thank you for your time much appreciated 💯
